packages/happy-dom/src/xml-http-request/utilities/XMLHttpRequestURLUtility.ts

```diff
@@ -1,4 +1,78 @@
-import { URL } from 'url';
+import { URL, UrlObject } from 'url';
+import Path from 'path';
+
+// MIME type.
+const LOCAL_MIME_TYPES = {
+	json: 'application/json',
+	xml: 'application/xml',
+	html: 'text/html',
+	text: 'text/plain',
+	xhtml: 'application/xhtml+xml',
+	xht: 'application/xhtml+xml',
+	xsl: 'application/xml',
+	xslt: 'application/xml',
+	rss: 'application/rss+xml',
+	atom: 'application/atom+xml',
+	yaml: 'application/x-yaml',
+	pdf: 'application/pdf',
+	zip: 'application/zip',
+	gzip: 'application/gzip',
+	rar: 'application/x-rar-compressed',
+	'7z': 'application/x-7z-compressed',
+	exe: 'application/x-msdownload',
+	csv: 'text/csv',
+	ics: 'text/calendar',
+	rtf: 'text/rtf',
+	js: 'application/javascript',
+	css: 'text/css',
+	png: 'image/png',
+	jpg: 'image/jpeg',
+	jpeg: 'image/jpeg',
+	gif: 'image/gif',
+	bmp: 'image/bmp',
+	ico: 'image/x-icon',
+	tiff: 'image/tiff',
+	tif: 'image/tiff',
+	svg: 'image/svg+xml',
+	svgz: 'image/svg+xml',
+	webp: 'image/webp',
+	wav: 'audio/wav',
+	webm: 'video/webm',
+	mp4: 'video/mp4',
+	mpeg: 'video/mpeg',
+	mpg: 'video/mpeg',
+	mov: 'video/quicktime',
+	avi: 'video/x-msvideo',
+	flv: 'video/x-flv',
+	mkv: 'video/x-matroska',
+	mka: 'audio/x-matroska',
+	m3u: 'audio/x-mpegurl',
+	m3u8: 'application/x-mpegURL',
+	pls: 'audio/x-scpls',
+	flac: 'audio/x-flac',
+	ogg: 'audio/ogg',
+	oga: 'audio/ogg',
+	ogv: 'video/ogg',
+	ogx: 'application/ogg',
+	opus: 'audio/opus',
+	spx: 'audio/ogg',
+	swf: 'application/x-shockwave-flash',
+	woff: 'font/woff',
+	woff2: 'font/woff2',
+	ttf: 'font/ttf',
+	eot: 'application/vnd.ms-fontobject',
+	otf: 'font/otf',
+	sfnt: 'application/font-sfnt',
+	bin: 'application/octet-stream',
+	doc: 'application/msword',
+	docx: 'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+	xls: 'application/vnd.ms-excel',
+	xlsx: 'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+	ppt: 'application/vnd.ms-powerpoint',
+	pptx: 'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+	odt: 'application/vnd.oasis.opendocument.text',
+	ods: 'application/vnd.oasis.opendocument.spreadsheet'
+};
```

Comment:
`mime-type` may use this library https://github.com/broofa/mime.

Reply:
well, I'm agree with you. 
Inevitably, this adds another third-party package, what do you think? @capricorn86

Reply:
I have a little bit of a bad experience with third-party packages. My suggestion is to leave it as you did it for now.

There are also plans to support both ESM soon (with a fallback for CommonJS), and third party packages can sometimes cause problems with ESM.

Reply:
I have a little bit of a bad experience with third party packages. My suggestion is to leave it as you did it.